(DOCSP-14303): Terminating and Reenabling Sync (#831)
mongodb/docs-realm: source/reference/terminating-and-reenabling-realm-sync.txt

.. _terminating-and-reenabling-sync:

====================================
Terminating & Re-enabling Realm Sync
====================================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

If you have :ref:`enabled Realm Sync <enable-sync>`, you may have to terminate and re-enable {+sync+} for one of the following reasons:

- An upgrade from a Shared Tier {+atlas-short+} Cluster to a Shared or Dedicated instance
- A move to an {+atlas-short+} cluster that is distributed across regions/cloud providers
- An upgrade to NVMe Atlas clusters
- An :manual:`oplog </core/replica-set-oplog/>` rollover

> - A paused {+sync+} session on a shared tier cluster due to in-frequent usage 

Reviewer: ```suggestion
- A paused {+sync+} session on a shared tier cluster due to infrequent usage 
```
infrequent should not be hyphenated, it's just a normal word

> - troubleshooting, at the request of MongoDB Support

Reviewer: ```suggestion
- Troubleshooting, at the request of MongoDB Support
```
Capitalize for parity with other bullet points

.. _terminating-and-reenabling-sync:

=================================

> Terminating/Reenabling Realm Sync

Reviewer: ```suggestion
Terminate Realm Sync
```
I figure that @nlarew will come after you with some kind of weapon if you add a _double_ gerund page to his toctree 😉

Author: The page title was actually a request from Drew. Given that this is nested within "Reference", I think forgoing the single gerund is fine in this case.

Reviewer: I personally feel like a slash and double gerund is pretty awkward for a page title -- after all, this page only discusses terminating sync, and just points you elsewhere to re-enable it -- but how about this:

"Terminating & Re-enabling Realm Sync"

Author: 👍

=================================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

If you have :ref:`enabled Realm Sync <enable-sync>`, you may have to terminate and reenable {+sync+} for one of the following reasons:

> - When upgrading from a Shared Tier {+atlas-short+} Cluster (either to a Shared or Dedicated instance)

Reviewer: Suggestion: keep bullet point wording consistent with the intro you established above "for one of the following reasons":
```suggestion
- An upgrade from a Shared Tier {+atlas-short+} Cluster to a Shared or Dedicated instance
```

> - When moving an {+atlas-short+} cluster across regions/cloud providers

Reviewer: Similarly,
```suggestion
- A move to an {+atlas-short+} cluster that is distributed across regions/cloud providers
```
Note: is there a proper name for this? I couldn't find one with a short search, but maybe the Atlas team knows.

Author: There is no proper name to describe moving a cluster to a different region - https://docs.atlas.mongodb.com/tutorial/move-cluster/

> - When upgrading NVMe Atlas clusters

Reviewer: ```suggestion
- Any upgrade to NVMe Atlas clusters
```

Author: Did you mean: `An upgrade to NVMe Atlas clusters`?

Reviewer: I actually did mean "any" in my suggestion but that works too, probably better tbh.

Author: done 👍

> - If the Oplog rolls over due to a Bug or misconfiguration

Reviewer: ```suggestion
- An :manual:`oplog </core/replica-set-oplog/>` rollover
```

Author: 👍

> - If {+sync+} becomes paused on a shared tier cluster with in-frequent usage 

Reviewer: ```suggestion
- A paused {+sync+} session on a shared tier cluster due to in-frequent usage 
```

Author: 👍

> - For certain troubleshooting steps (at the request of MongoDB Support)

Reviewer: ```suggestion
- troubleshooting, at the request of MongoDB Support
```

Author: 👍

.. _terminating-realm-sync:

Terminate Realm Sync 
--------------------

> Terminate {+sync+} in your {+backend+} application to stop syncing data across

Reviewer: Suggestion: You can actually use the {+app+} source constant here ("Realm App") to refer to an individual instance of the MongoDB Realm backend
```suggestion
Terminate {+sync+} in your {+app+} to stop syncing data across
```

Author: 👍

devices. 

> .. warning:: 

Reviewer: We should probably create a title for this warning -- maybe "Terminating Sync may Lose Unsynced Client Changes"

Author: Adding a title sounds good, but I don't think that suggestion captures everything in the note. How about: "Preventing loss of unsynced changes when terminating Realm Sync"?

Author: or more simply "Preventing loss of unsynchronized changes"

Reviewer: How about "Recovering Unsynchronized Changes after Terminating Sync"?

Author: That works for me! 👍

 
 Terminating/Re-enabling Sync will prevent local changes that you have not

>  synced yet from syncing. If you plan on terminating Sync in production and wish

Reviewer: two occurrences of "sync" in the last four words of a sentence feels repetitive to me. I would suggest something like:

```
Terminating and re-enabling {+sync+} prevents unsynchronized
client changes from automatically syncing. To recover any
unsychronized changes, implement a manual client reset
that handles this case in your client applications:
```

Author: 👍
